src/activity_handlers.cpp

```diff
@@ -2538,7 +2539,41 @@ void activity_handlers::craft_do_turn( player_activity *act, player *p )
 {
     const recipe &rec = recipe_id( act->name ).obj();
     float crafting_speed = p->crafting_speed_multiplier( rec, true );
-    if( crafting_speed <= 0.0f ) {
+    bool changed_light = false;
+
+    if( p->lighting_craft_speed_multiplier( rec ) <= 0.0f &&
+        g->m.veh_at( p->pos() ).has_value() &&
```

Comment:
You don't need to call `has_value()` here, `optional` supports conversion to bool in this situation.